android/src/toga_android/widgets/webview.py

```diff
@@ -23,14 +28,40 @@ def onReceiveValue(self, value):
         self.result.set_result(res)
 
 
+class TogaNavigationEvent:
+    def __init__(self, webresourcerequest):
+        self.request = webresourcerequest
+        self.cancel = False
+
+
+class TogaWebClient(static_proxy(WebViewClient)):
+    def __init__(self, impl):
+        super().__init__()
+        self.webview_impl = impl
+
+    @Override(jboolean, [A_WebView, WebResourceRequest])
+    def shouldOverrideUrlLoading(self, webview, webresourcerequest):
+        if self.webview_impl.interface.on_navigation_starting:
+            event = TogaNavigationEvent(webresourcerequest)
+            allow = self.webview_impl.interface.on_navigation_starting(
+                webresourcerequest.getUrl().toString()
+            )
```

Comment:
I haven't checked, but this could be a problem if on_navigation_starting is an async callback, because the returned value will be a future.